dags/synapse-datasets-created-7-days.py

```diff
@@ -0,0 +1,221 @@
+            created_by,
+            CONCAT(first_name,' ',last_name) AS user_name,
+            is_public,
+        FROM 
+            synapse_data_warehouse.synapse.node_latest as n
+        LEFT JOIN 
+            synapse_data_warehouse.synapse.userprofile_latest as u
+            ON 
+            n.created_by = u.id
+        WHERE 
+            (
+                node_type IN ('dataset', 'project') 
+                OR 
+                (content_type = 'dataset')
+            )
+        AND 
+            (
+                entity_created_date >= DATEADD(DAY, -7, '{query_date or context["params"]["current_date_time"]}')
+            )
+        ORDER BY entity_created_date;
+        """
+        print(query)
+        cs.execute(query)
+        created_entity_df = cs.fetch_pandas_all()
+
+        entity_created = []
+        for _, row in created_entity_df.iterrows():
+            entity_created.append(
+                EntityCreated(
+                    name=row["NAME"],
+                    id=row["ID"],
+                    node_type=row["NODE_TYPE"],
+                    parent_id=row["PARENT_ID"],
+                    content_type=row["CONTENT_TYPE"],
+                    created_on=row["ENTITY_CREATED_DATE"],
+                    user_name=row["USER_NAME"],
+                    created_by=row["CREATED_BY"],
+                    is_public=row["IS_PUBLIC"],
+                )
+            )
+        return entity_created
+
+    @task
+    def generate_slack_message(entity_created: List[EntityCreated], **context) -> str:
+        """Generate the message to be posted to the slack channel."""
+        message = ":synapse: Datasets or projects created in the last 7 days \n\n"
+        for index, row in enumerate(entity_created):
+            if row.content_type:
+                data_type = row.content_type
+            else:
+                data_type = row.node_type
+            message += f"{index+1}. <https://www.synapse.org/#!Synapse:syn{row.id}|*{row.name}*> (Type: {data_type}, Created on: {row.created_on}, Created by: <https://www.synapse.org/Profile:{row.created_by}/profile|{row.user_name}>, Public: {row.is_public})\n\n"
+        return message
+
+    @task
+    def post_slack_messages(message: str) -> bool:
+        """Post the top downloads to the slack channel."""
+        client = WebClient(token=Variable.get("SLACK_DPE_TEAM_BOT_TOKEN"))
+        result = client.chat_postMessage(channel="hotdrops", text=message)
+        print(f"Result of posting to slack: [{result}]")
+        return result is not None
+
+    @task
+    def push_results_to_synapse_table(
+        entity_created: List[EntityCreated], **context
+    ) -> None:
+        """Push the results to a Synapse table."""
+        data = []
+        # convert context["params"]["backfill_date_time"] to date in same format as date.today()
+        today = (
+            date.today()
+            if not context["params"]["backfill"]
+            else datetime.strptime(
```

Comment:
@BWMac if I set the back fill date parameter in airflow like how I set it up now, by default is using the UTC time zone. But when the table gets uploaded to synapse, synapse would use the user's local time zone. Is that okay?

Reply:
Yes, that is how the other DAGs work. See this explanation in one of the module docstrings [here](https://github.com/Sage-Bionetworks-Workflows/orca-recipes/blob/57a03fa128d9365be239d0ceda4076fc7647b135/dags/top-public-synapse-projects-from-snowflake.py#L5-L10)

Reply:
Got it. Thanks. Should I also add that docstring that you link to this file?

Reply:
Yeah that's probably a good idea!

Reply:
Got it. I will add the docstring.